### ISSUE TITLE
Ambiguous 'machinectl list' exit code

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#5611]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### systemd version the issue has been seen with

>systemd 232
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN

*NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!*

### Used distribution

>Antergos Linux release 16.11 (ISO-Rolling)

### In case of bug report: Expected behaviour you didn't see

>Currently running containers to be listed and the exit code, 0.

### In case of bug report: Unexpected behaviour you saw

>Currently running containers, but the exit code 1.

### In case of bug report: Steps to reproduce the problem

>machinectl list
```
MACHINE  CLASS     SERVICE        OS   VERSION ADDRESSES
chromium container systemd-nspawn arch -       -
hexchat  container systemd-nspawn arch -       -

2 machines listed.
```
>echo $?
```
1
```
Someone I talked to on systemd irc channel told me that [this](https://github.com/systemd/systemd/blob/v232/src/machine/machinectl.c#L330) and [this](https://github.com/systemd/systemd/blob/v233/src/machine/machinectl.c#L331) are sections of the code responsible for the exit code of 1 (non-zero failure code).
In his opinion (and I agree)

> it's not right to be exiting with EXIT_FAILURE when there are no addresses, but the implementation is creating an ambiguity not differentiating between an error and no addresses defined

**Comment by evverx** [comment]
Could you please apply the patch from https://github.com/systemd/systemd/pull/5587? Does it fix the bug?

**Comment by ghost** [comment]
@evverx I'm so terribly sorry but I don't patch and I don't build. Hope you won't mind.

**Comment by keszybz** [comment]
I'm pretty sure this is fixed by #5587.
